tests/test_expression.py

```diff
@@ -228,6 +228,27 @@ def test_add_simplified(self) -> None:
             Add(Variable("X"), Variable("Y")),
         )
 
+        # 3.simplified(1)
+        self.assertEqual(
+            Sub(Variable("X"), Variable("Y")).simplified({Variable("X"): Variable("Z")}),
+            Add(Variable("Z"), -Variable("Y")),
+        )
+
+        # 3.simplified(1).simplified(2) = 4
+        self.assertEqual(
+            Sub(Variable("Z"), Variable("Y")).simplified(
+                {Variable("Z"): Add(Variable("Y"), Variable("Q"))}
+            ),
+            Add(Add(Variable("Y"), Variable("Q")), -Variable("Y")),
+        )
+
+        # 4.simplified() = TRUE
+        self.assertTrue(
+            Equal(Variable("Q"), Sub(Add(Variable("Y"), Variable("Q")), Variable("Y")))
+            .simplified()
+            .simplified()
+        )
```

Comment:
The tests mainly test the `Sub` expression so they should be moved to `test_sub_simplified`.